src/teamspeak/mod.rs

```diff
@@ -241,88 +282,90 @@ impl TeamSpeakConnection {
     }
 
     pub async fn set_nickname(&mut self, name: String) {
-        self.handle
+        if let Err(e) = self
+            .handle
+            .as_mut()
+            .expect("connect_for_bot was called")
             .with_connection(move |mut conn| {
                 conn.get_state()
                     .expect("can get state")
                     .client_update()
                     .set_name(&name)
                     .send(&mut conn)
-                    .map_err(|e| error!("Failed to set nickname: {}", e))
             })
             .await
-            .unwrap()
-            .unwrap();
+            .and_then(|v| v)
+        {
+            error!(self.logger, "Failed to set nickname: {}", e);
+        }
     }
 
     pub async fn set_description(&mut self, desc: String) {
-        self.handle
+        if let Err(e) = self
+            .handle
+            .as_mut()
+            .expect("connect_for_bot was called")
             .with_connection(move |mut conn| {
                 let state = conn.get_state().expect("can get state");
-                let _ = state
+                state
                     .clients
                     .get(&state.own_client)
                     .expect("can get myself")
                     .edit()
                     .set_description(&desc)
                     .send(&mut conn)
-                    .map_err(|e| error!("Failed to change description: {}", e));
             })
             .await
-            .unwrap()
+            .and_then(|v| v)
```

Comment:
big if, not sure if we can make these better, maybe get better at propagating errors?

Reply:
Yes, in another PR.